### PR TITLE
Player Tags

### PR DESCRIPTION
**Pull request by ah1053**
1. Removing datagram module from package.json, and its dependencies from package-lock.json since it is included by default in node 12 core modules.
2. Updated readme to use the npm module.
3. Added PlayerTags (optional custom data can be added by the client to the player)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,23 +1,26 @@
...
-  "version": "1.0.0",
+  "version": "1.0.1",
   "description": "A sample server for matchmaking & a relay server for multiplayer experiences",
   "main": "server.js",
   "scripts": {
     "start": "node server.js",
     "dev": "npx nodemon server.js"
   },
-  "keywords": [],
+  "keywords": [
```
Comment (lgarner) [inline package.json:10]: Ah, search terms.

File: server.js [hunk 2]
```diff
@@ -81,11 +82,11 @@ wsServer.on('connection', function connection(ws, request) {
                     break
                 case 'CreateRoom':
                     console.log('CreateRoom Request received from client %o', jsonObj)
-                    roomId = util.createRoom(lobby, jsonObj.playerId, jsonObj.playerName, jsonObj.playerAvatar, jsonObj.maxPlayersPerRoom)
+                    roomId = util.createRoom(lobby, jsonObj.playerId, jsonObj.playerName, jsonObj.playerAvatar, jsonObj.maxPlayersPerRoom, jsonObj.playerTags)
```
Comment (lgarner) [inline server.js:85]: App dependent tags?

Comment (ah1053) [reply server.js:85]: Exactly, When the player connects to the server, they can pass tags <KeyValue List> to the server.
The Hashtable to Dictionary in DistributedMatchingEngine.cs (tags) made the c# implementation super easy for me. 
https://github.com/mobiledgex/edge-multiplay-unity-client/pull/9

File: models/Player.js [hunk 3]
```diff
@@ -26,12 +26,14 @@ class Player {
      * @param  {string} playerName name of the player sat by the client on JoinRoomRequest or the CreateRoomRequest
      * @param  {integer} playerAvatar the avatar of the player sat by the client on JoinRoomRequest or the CreateRoomRequest
      * @param  {integer} playerIndex the order of the player in the room
+     * @param  {object} playerTags KeyValue Pair Array of custom data associated with the player
      */
-    constructor( playerId, playerName, playerAvatar, playerIndex) {
+    constructor( playerId, playerName, playerAvatar, playerIndex, playerTags) {
```
Comment (franklin-huang-mobiledgex) [inline models/Player.js:31]: nit: extra space between "(" and "playerId"

File: server.js [hunk 4]
```diff
@@ -67,7 +67,8 @@ wsServer.on('connection', function connection(ws, request) {
             switch (jsonObj.type) {
                 case 'JoinOrCreateRoom':
                     console.log('JoinOrCreateRoom Request received from client %o', jsonObj)
-                    roomId = util.joinOrCreateRoom(lobby, jsonObj.playerId, jsonObj.playerName, jsonObj.playerAvatar, jsonObj.maxPlayersPerRoom)
+
```
Comment (franklin-huang-mobiledgex) [inline server.js:70]: nit: extra line

File: server.js [hunk 5]
```diff
@@ -67,7 +67,8 @@ wsServer.on('connection', function connection(ws, request) {
             switch (jsonObj.type) {
                 case 'JoinOrCreateRoom':
```
Comment (franklin-huang-mobiledgex) [inline server.js:68]: I'm a little bit confused why there is a JoinOrCreateRoom when there are also JoinRoom and CreateRoom cases as well. What does this case add?

File: server.js [hunk 6]
```diff
@@ -194,12 +195,12 @@ function addToLobby(connection){
...
+ * @param  {Object} connection object contains the request object, socket object and headers object.
  */
 function rejectConnection(connection){
     var {socket} = connection
     console.log('destroying connection')
     socket.destroy()
 }
 
-module.exports = {wsServer, addToLobby, rejectConnection}
+module.exports = {wsServer, addToLobby, rejectConnection}
```
Comment (franklin-huang-mobiledgex) [inline server.js:206]: nit: add new line at end of file to get rid of red circle

**Review by franklin-huang-mobiledgex — APPROVED** [review]
Looks good!

File: server.js (hunk 5 above)
Comment (ah1053) [reply server.js:68]: This usually added for dynamically creating room or joining available rooms
It's useful for light games like pingpong where you want to just play the game with anyone available

**Comment by ah1053** [comment]
@LeonAdams  please let me know once you try it out, so I can publish the package to npm
https://mobiledgex.atlassian.net/browse/EDGECLOUD-4596

**Review by LeonAdams — APPROVED** [review]
Looks good. Checking the video now

**Review by brucearmstrong — APPROVED** [review]

**Pull request merged**